lib/vandelay/util/cache.rb

```diff
@@ -3,7 +3,22 @@
 module Vandelay
   module Util
     class Cache
-      # your code here
+      def initialize
+        @redis = Redis.new(url: Vandelay.config.dig("persistence", "redis_url"))
+      end
+
+      def read(key)
+        @redis.get key
+      end
+
+      def write(key, data, expiry = 10 * 60)
+        @redis.set key, data
+        @redis.expire key, expiry
+      end
+
+      def self.instance
```

Comment:
I decided to make this class a singleton to ensure only one instance of cache is available in entire app.